### PR TITLE
Added methods to set default namespace (onprem only)

### PR DESCRIPTION
**Pull request by connelly38**
Methods are hidden for now.
Requires onprem KV version 22.3, patch revision 24 or higher.

### CODE REVIEW

**Review by cezarfx — CHANGES_REQUESTED** [review]

File: driver/src/main/java/oracle/nosql/driver/NoSQLHandleConfig.java [hunk 1]
```diff
@@ -1074,6 +1086,40 @@ public String getDefaultCompartment() {
         return compartment;
     }
 
+    /**
+     * @hidden
+     *
+     * Onprem use only.
```
Comment (gmfeinberg) [inline driver/src/main/java/oracle/nosql/driver/NoSQLHandleConfig.java:1092]: "On-premise only"

File: driver/src/main/java/oracle/nosql/driver/NoSQLHandleConfig.java [hunk 2]
```diff
@@ -1074,6 +1086,40 @@ public String getDefaultCompartment() {
         return compartment;
     }
 
+    /**
+     * @hidden
```
Comment (gmfeinberg) [inline driver/src/main/java/oracle/nosql/driver/NoSQLHandleConfig.java:1090]: Why is this (and corresponding get) hidden?

**Comment by gmfeinberg** [comment]
Shouldn't there be per-request methods as well, similar to compartment?

File: driver/src/main/java/oracle/nosql/driver/NoSQLHandleConfig.java [hunk 3]
```diff
@@ -205,14 +205,14 @@ public class NoSQLHandleConfig implements Cloneable {
...
      * The default namespace to use for all requests. If this is null (the
      * default), no namespace is used unless specified in table names in
      * requests and SQL statements.
      *
-     * Any namespace specified in a table name (using the namespace:tablename
-     * syntax) in requests and/or SQL statements overrides this default.
+     * Any non-namespace qualified table name in requests and/or SQL
+     * statements will be resolved/qualified to the specified namespace.
      */
     private String namespace;
```
[the remark this answers is not in the document]
Comment (connelly38) [reply driver/src/main/java/oracle/nosql/driver/NoSQLHandleConfig.java]: Good idea. Done.

Comment (cezarfx) [inline driver/src/main/java/oracle/nosql/driver/NoSQLHandleConfig.java]: Rename to defaultNamespace for clarity.

Comment (connelly38) [reply driver/src/main/java/oracle/nosql/driver/NoSQLHandleConfig.java]: I did this already....

**Review by cezarfx — APPROVED** [review]
Looks good and per request defaultNamespace will be added in a different PR.

**Pull request merged**